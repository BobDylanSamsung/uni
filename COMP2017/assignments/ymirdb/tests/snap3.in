set a 1 2 3
set b 1 a 2
snapshot
set a 4 5 6
snapshot
set c a b
snapshot
list snapshots
forward a
forward b
forward c
backward a
backward b
backward c
checkout 1
forward a
forward b
forward c
backward a
backward b
backward c
rollback 2
list entries
list snapshots
rollback 1
list entries
list snapshots
set d 1 2 3 4 5
snapshot
list snapshots
bye
