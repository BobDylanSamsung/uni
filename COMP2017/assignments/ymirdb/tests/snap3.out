ok

ok

saved as snapshot 1

ok

saved as snapshot 2

ok

saved as snapshot 3

3
2
1

nil

a

a, b

b, c

c

nil

ok

nil

a

no such key

b

nil

no such key

ok

b [1 a 2]
a [4 5 6]

2
1

ok

b [1 a 2]
a [1 2 3]

1

ok

saved as snapshot 2

2
1

bye
